File: anul2/semestrul1/map/lab2/sort/SortingTask.java
package sort;

import model.Task;

public class SortingTask extends Task {
    private int[] array;
    private SortingStrategy strategy;
    private AbstractSorter taskSorter;

    public SortingTask(String taskID, String descriere, SortingStrategy strategy, int[] array) {
        super(taskID, descriere);
        this.strategy = strategy;
        this.array = array;
        //Obtin o instanta a fabricii de sortare si creez un sorter specific strategiei
        SortingFactory factory = SortingFactory.getInstance();
        this.taskSorter = factory.makeSorter(strategy);
    }

    @Override
    public void execute() {
        taskSorter.sort(array);
    }
}

File: Third Semester - Advanced Programming Methods/Lab10/sort/SortingTask.cs
namespace lab10.sort;
using Task = lab10.model.Task;
public class SortingTask: Task 
{
    private int[] _array;
    private SortingStrategy _sortingStrategy;
    private AbstractSorter _abstractSorter;

    public SortingTask(String taskID, String description, SortingStrategy sortingStrategy, int[] array): base(taskID, description)
    {
        this._array = array;
        this._sortingStrategy = sortingStrategy;
        SortingFactory factory = SortingFactory.Instance;
        this._abstractSorter = factory.makeSorter(sortingStrategy);
    }

    public override void execute()
    {
        _abstractSorter.sort(_array);
    }
}